src/test/java/com/exactpro/th2/crawler/CrawlerTest.java

```diff
@@ -187,4 +190,48 @@ public void dataServiceFail() throws IOException {
 
         Assertions.assertThrows(RuntimeException.class, crawler::process, exceptionMessage);
     }
+
+    @Test
+    @DisplayName("Crawler gets extra aliases while processing intervals")
+    public void dynamicAliasesProcessing() throws IOException, UnexpectedDataProcessorException {
+        CrawlerConfiguration configuration = CrawlerManager.createConfig(
+                "2021-06-16T12:00:00.00Z", DataType.MESSAGES, Set.of("alias\\d+"), SessionAliasType.REGEXP
+        );
+
+        CrawlerManager manager = new CrawlerManager(configuration);
+
+        List<Message> messages1 = MessageReaderKt.readMessages(Paths.get("src/test/resources/dynamic_aliases_messages-1.txt"));
+        List<Message> messages2 = MessageReaderKt.readMessages(Paths.get("src/test/resources/dynamic_aliases_messages-2.txt"));
+
+        List<String> aliases1 = new ArrayList<>(List.of("alias1", "alias2"));
+        StringList list1 = StringList.newBuilder().addAllListString(aliases1).build();
+
+        List<String> aliases2 = new ArrayList<>(List.of("alias1", "alias2", "alias3"));
+        StringList list2 = StringList.newBuilder().addAllListString(aliases2).build();
+
+        when(manager.getDataProviderMock().getMessageStreams(any(Empty.class))).thenReturn(list1, list2);
+
+        Crawler crawler = manager.createCrawler();
+
+        Iterator<StreamResponse> iterator1 = new MessageSearchResponse(messages1).iterator();
+        Iterator<StreamResponse> iterator2 = new MessageSearchResponse(messages2).iterator();
+
+        when(manager.getDataProviderMock().searchMessages(any(MessageSearchRequest.class))).thenReturn(iterator1, iterator2);
```

Comment:
I think you should configure stub more specific (expect the certain aliases in the search request)

Reply:
> expect the certain aliases in the search request

What do you mean by that? To add a `verify()` call to check what aliases Crawler gets from each invocation of `searchMessages()`?

Reply:
I mean you should specify when mock should return each iterator. The current version will return the first iterator and then the second. It won't look at MessageSearchRequest parameters. I suggest you use `argThat` instead of `any` for mocking and check inside the `argThat`method which aliases were requested